addons/web/static/src/views/fields/signature/signature_field.js

```diff
@@ -135,7 +136,10 @@ export class SignatureField extends Component {
 
 export const signatureField = {
     component: SignatureField,
-    fieldDependencies: [{ name: "write_date", type: "datetime" }],
+    fieldDependencies: [
+        { name: "write_date", type: "datetime" },
+        { name: "display_name", type: "string" },
```

Comment:
@aab-odoo  If I try  clicking on the widget I get
```
Cannot read properties of undefined (reading 'type')
    at SignatureField.onClickSignature
```
on
```js
const fullNameData = record.data[fullName];
if (record.fields[fullName].type === "many2one") {
``` because display_name is not there. Am I doing something wrong then?

Reply:
I guess the field `display_name` could be added in the view where the field is used with attribute `full_name="display_name`.

`fieldDependencies` isn't a great concept, because it shadows important information, in particular for the onchange mechanism, by allowing people to not specify some fields in the arch as they are enforced by the js. We might consider removing it at some point.

But this part isn't related to the scope of this PR right? I guess you also have the crash in master (and sooner?)

Reply:
For some reason, the other view that uses the widget fetches the display_name even thought it is not set as a field in the view. I made the changes and it's working now without the fieldDependencies! Thanks